x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/utils.ts

```diff
@@ -215,13 +221,61 @@ export const transformBulkError = (
   }
 };
 
-export const getIndex = (getSpaceId: () => string, config: LegacyServices['config']): string => {
-  const signalsIndex = config().get<string>(`xpack.${APP_ID}.${SIGNALS_INDEX_KEY}`);
-  const spaceId = getSpaceId();
+export const buildRouteValidation = <T = {}>(schema: Joi.Schema): RouteValidationFunction<T> => (
+  payload: T,
+  { ok, badRequest }
+) => {
+  const { value, error } = schema.validate(payload);
+  if (error) {
+    return badRequest(error.message);
+  }
+  return ok(value as T); // TODO: infer type from our schema
```

Comment:
I took this `as cast` away from here and didn't see any type issues. I just did this:

```ts
return ok(value);
```